### ISSUE TITLE
regen.sh on osx

### ISSUE THREAD

**Issue by awalterschulze** [golang/protobuf#196]
Could the regen.sh script possibly be changed to also work on osx?

```
./regen.sh: line 44: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
```

**Comment by mwmahlberg** [comment]
Hm, first of all, as per official docs, there is [no **`-e`** option for bash ](http://linux.die.net/man/1/bash), which makes me curious. And `declare` is a bash built-in command. Every halfway recent version of bash should support it.

The funny thing is that the declare function of `sh` has options mentioned above (see manpage). So my first guess is that for some weird reason, regex.sh is run with the sh interpreter rather than bash – however that may happen.

**Comment by mwmahlberg** [comment]
As a matter of fact, it might be just the opposite around. Not sure, but changing the shebang to

    #!/bin/sh

might actually solve the problem.

**Comment by awalterschulze** [comment]
I still get the same error

**Comment by mwmahlberg** [comment]
The same one? Ok, I will look into it to get it running. Any way I can verify the results?

**Comment by awalterschulze** [comment]
I tried this regen diff
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -e
+#!/bin/sh -e
 #
 # This script fetches and rebuilds the "well-known types" protocol buffers.
 # To run this you will need protoc and goprotobuf installed;
@@ -39,7 +39,7 @@ echo 1>&2 "base: $base"
 cd $base

 echo 1>&2 "fetching latest protos... "
-git clone -q $UPSTREAM $tmpdir
+git clone $UPSTREAM $tmpdir
 # Pass 1: build mapping from upstream filename to our filename.
 declare -A filename_map
 for f in $(cd $PKG && find * -name '*.proto'); do
```

I have this declare
```
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
```

Don't you have OSX?

**Comment by mwmahlberg** [comment]
Aye, I am already working on it. It is a bit strange. Not sure what happens there.

**Comment by mwmahlberg** [comment]
@awalterschulze After banging my head against the wall for a few hours, I decided that I would rather [reimplement the script](https://gist.github.com/mwmahlberg/671badadc80e0498213bae0d9e66ab6a) than debug it any further. As far as I can see from the sources, upstream now holds an `option go_package…;`, hence the script does not change the original package name. Running a `go test` after calling `regen2.sh` worked. I have to test it against Linux. But please let me first know wether this works for you.

**Comment by awalterschulze** [comment]
I removed the -q for the git clone 

```
git clone $UPSTREAM $tmpdir
```

and I added a -e to the top
```
#!/bin/bash -e
```

But then it was working :)

I hope they take it in a pull request.
Otherwise if you can minimize the diff it would be nice to merge this into gogoprotobuf.

**Comment by mwmahlberg** [comment]
Very, very strange. As said, there is no `-e` option for bash. How are you calling that script?

And I am not quite sure what you mean with "minimize the diff". Could you explain? ;)

**Comment by awalterschulze** [comment]
I am calling it like so
```
./regen.sh
```
I think it does the same as
```
set -e
```

"set -e causes the shell to exit if any subcommand or pipeline returns a non-zero status"

**Comment by awalterschulze** [comment]
Minimizing the diff:  If you can make the diff between the golang/protobuf regen.sh and your new regen.sh as small as possible that will make it easier to maintain when golang/protobuf edits its own regen.sh.  Obviously merging in a pull request into golang/protobuf is first prize.

**Comment by mwmahlberg** [comment]
Uh, well, I built the new script from ground up, so minimizing the diff actually might become very hard. My suggestion would be to make a PR against your ptypes branch where we can test and polish it as needed, and when we are done you can do a PR for golang/protobuf?

**Comment by awalterschulze** [comment]
You don't want to be in that place, because golang/protobuf will never take a pull request from gogoprotobuf.  Or at least I have tried in the past and have failed.
If you can get this merged into golang/protobuf that would be a huge help.  I think this change is small enough for them to consider.

**Comment by mwmahlberg** [comment]
Aye, will polish it a bit. *sigh* You want a PR against `ptypes` anyway?

**Comment by awalterschulze** [comment]
No the ptypes branch is a mess at the moment.  I'll see what I can salvage there at some point.

**Comment by mwmahlberg** [comment]
Hm, there is already #144 😲 Have to dig into that.

**Comment by awalterschulze** [comment]
Start a new one

**Comment by mwmahlberg** [comment]
I am at it. Though I need to do some testing on the various flavors of the respective tools. At least FreeBSD, Linux and Solaris. OS X seems to work 😏

**Comment by jhump** [comment]
This should be fixed as of #412.

**Comment by dsnet** [comment]
Closing as fixed (I did not verify) according to @jhump's comment. If it occurs again, we can file another issue.
